src/main/java/org/openstreetmap/atlas/geography/atlas/change/FeatureChangeMergingHelpers.java

```diff
@@ -654,35 +605,55 @@ private static FeatureChange mergeRelations(final FeatureChange left, final Feat
-                MemberMergeStrategies.diffBasedLongMerger);
+        final MergedMemberBean<RelationBean> mergedMembersBean = new MemberMerger.Builder<RelationBean>()
+                .withMemberName("relationMembers").withBeforeEntityLeft(beforeEntityLeft)
+                .withAfterEntityLeft(afterEntityLeft).withBeforeEntityRight(beforeEntityRight)
+                .withAfterEntityRight(afterEntityRight)
+                .withMemberExtractor(entity -> ((Relation) entity).members() == null ? null
+                        : ((Relation) entity).members().asBean())
+                .withAfterViewNoBeforeMerger(MemberMergeStrategies.simpleRelationBeanMerger)
+                .withAfterViewConsistentBeforeMerger(
+                        MemberMergeStrategies.diffBasedRelationBeanMerger)
+                .withAfterViewConflictingBeforeMerger(
+                        MemberMergeStrategies.conflictingBeforeViewRelationBeanMerger)
+                .withBeforeViewMerger(MemberMergeStrategies.beforeViewRelationBeanMerger).build()
+                .mergeMember();
+
+        final MergedMemberBean<Set<Long>> mergedAllRelationsWithSameOsmIdentifierBean = new MemberMerger.Builder<Set<Long>>()
+                .withMemberName("allRelationsWithSameOsmIdentifier")
+                .withBeforeEntityLeft(beforeEntityLeft).withAfterEntityLeft(afterEntityLeft)
+                .withBeforeEntityRight(beforeEntityRight).withAfterEntityRight(afterEntityRight)
+                .withMemberExtractor(atlasEntity -> ((Relation) atlasEntity)
+                        .allRelationsWithSameOsmIdentifier() == null
+                                ? null
+                                : ((Relation) atlasEntity).allRelationsWithSameOsmIdentifier()
+                                        .stream().map(Relation::getIdentifier)
+                                        .collect(Collectors.toSet()))
+                .withAfterViewNoBeforeMerger(MemberMergeStrategies.simpleLongSetMerger)
+                .withAfterViewConsistentBeforeMerger(MemberMergeStrategies.diffBasedLongSetMerger)
+                .build().mergeMember();
+
+        final MergedMemberBean<RelationBean> mergedAllKnownMembersBean = new MemberMerger.Builder<RelationBean>()
+                .withMemberName("allKnownOsmMembers").withBeforeEntityLeft(beforeEntityLeft)
+                .withAfterEntityLeft(afterEntityLeft).withBeforeEntityRight(beforeEntityRight)
+                .withAfterEntityRight(afterEntityRight)
+                .withMemberExtractor(entity -> ((Relation) entity).allKnownOsmMembers() == null
+                        ? null : ((Relation) entity).allKnownOsmMembers().asBean())
+                .withAfterViewNoBeforeMerger(MemberMergeStrategies.simpleRelationBeanMerger)
+                .withAfterViewConsistentBeforeMerger(
+                        MemberMergeStrategies.diffBasedRelationBeanMerger)
+                .withAfterViewConflictingBeforeMerger(
```

Comment:
Again, do we need to resolve conflicting beforeViews here? If `allKnownOsmMembers` is consistent across shards, then I think not.

Reply:
Same I think...